web/packages/core/src/ruffle-player.ts

```diff
@@ -667,6 +669,38 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * Async helper function for fetching the SWF.
+     */
+    private async fetchSWF(): Promise<void> {
+        const response = await fetch(this.swfUrl!);
+        if (!response.ok) {
+            console.error("SWF download failed");
+            return;
+        }
+        const blob = await response.blob();
+        const blobUrl = window.URL.createObjectURL(blob);
+        const swfDownloadA = document.createElement("a");
+        swfDownloadA.style.display = "none";
+        swfDownloadA.href = blobUrl;
+        swfDownloadA.download = this.swfUrl!.replace(/^.*\//, "");
```

Comment:
This can be done similarly to:
https://github.com/ruffle-rs/ruffle/blob/df514f319a11c770310b8260a68f020af2061b80/web/packages/extension/src/player.ts#L21